libs/domain/checkout/address/address.component.spec.ts

```diff
@@ -154,34 +155,67 @@ describe('CheckoutAddressComponent', () => {
       beforeEach(async () => {
         addressService.getAddresses.mockReturnValue(of([{}]));
-        element = await fixture(
-          html`<oryx-checkout-address></oryx-checkout-address>`
-        );
       });
 
-      it('should not render the oryx-user-address-form', () => {
-        expect(element).not.toContainElement('oryx-user-address-form');
-      });
+      describe('and the enableList option is false', () => {
+        beforeEach(async () => {
+          element = await fixture(
+            html`<oryx-checkout-address
+              .options=${{ enableList: false } as CheckoutAddressOptions}
+            ></oryx-checkout-address>`
+          );
+        });
+
+        it('should not render the oryx-user-address-form', () => {
+          expect(element).not.toContainElement('oryx-user-address-form');
+        });
+
+        it('should not render the oryx-user-address-list', () => {
+          expect(element).not.toContainElement('oryx-user-address-list');
+        });
 
-      it('should render the oryx-user-address-list', () => {
-        expect(element).toContainElement('oryx-user-address-list');
+        it('should render the oryx-user-address', () => {
+          expect(element).toContainElement('oryx-user-address');
+        });
       });
 
-      describe('and an change event is dispatched on the list', () => {
-        beforeEach(() => {
-          element.dispatchEvent = vi.fn();
-          const list = element.renderRoot.querySelector(
-            'oryx-user-address-list'
-          ) as HTMLElement;
-          list.dispatchEvent(
-            new CustomEvent('change', {
-              detail: { address: { foo: 'bar' } },
-            })
+      describe('and the enableList option is true', () => {
+        beforeEach(async () => {
+          element = await fixture(
+            html`<oryx-checkout-address
+              .options=${{ enableList: true } as CheckoutAddressOptions}
+            ></oryx-checkout-address>`
           );
         });
 
-        it('should set the address to the state service', () => {
-          expect(addressStateService.set).toHaveBeenCalled();
+        it('should not render the oryx-user-address-form', () => {
+          expect(element).not.toContainElement('oryx-user-address-form');
+        });
+
+        it('should render the oryx-user-address-list', () => {
+          expect(element).toContainElement('oryx-user-address-list');
+        });
+
+        it('should not render the oryx-user-address', () => {
+          expect(element).not.toContainElement('oryx-user-address');
+        });
+
+        describe('and an change event is dispatched on the list', () => {
+          beforeEach(() => {
+            element.dispatchEvent = vi.fn();
+            const list = element.renderRoot.querySelector(
+              'oryx-user-address-list'
+            ) as HTMLElement;
+            list.dispatchEvent(
+              new CustomEvent('change', {
```

Comment:
And according to our convention it is public event and it should have `oryx` prefix, ie `oryx.change`

Reply:
yeah, we started to drop that convention. We could discuss it, but it's no so much part of this PR

Reply:
Won't they then overlap with standard HTML events?